File: app/src/main/java/com/jnjcomu/edison/storage/Configuration.java
package com.jnjcomu.edison.storage;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;

import com.jnjcomu.edison.encrypt.Encrypter;

import org.json.JSONObject;

import java.io.InputStream;

/**
 * Configuration Class
 *
 * 키스토어에 의해 암호화된 키를 해독하는 클레스.
 *
 *
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-13
 */
public class Configuration {
    private static final String TAG = "Configuration";
    private static final String ENCODING = "UTF-8";
    private static final String CONFIGURATION_JSON_NAME = "Edison.json";

    private JSONObject json;
    private String loplatId, loplatPw, hostUrl;

    public Configuration(Context context) {
        InputStream inputStream = null;
        Encrypter encrypt = new Encrypter(context);

        try {
            // read edison.json file
            inputStream = context.getAssets().open(CONFIGURATION_JSON_NAME);

            int size = inputStream.available();
            byte[] buffer = new byte[size];

            inputStream.read(buffer);
            String readString = new String(buffer, ENCODING);

            // get json object
            json = new JSONObject(readString);

            // get decrypted key
            JSONObject loplatJson = json.getJSONObject("loplat");

            String encryptHostUrl = json.getString("host_url");
            String encryptLoplatId = loplatJson.getString("id");
            String encryptLoplatPw = loplatJson.getString("pw");

            hostUrl = encrypt.decrypt(encryptHostUrl);
            loplatId = encrypt.decrypt(encryptLoplatId);
            loplatPw = encrypt.decrypt(encryptLoplatPw);
        } catch (Exception ignored) {
            Log.d(TAG, "Cant read json from assets");
        } finally {
            try {
                inputStream.close();
            } catch (Exception ignored) {
                Log.d(TAG, "Fail to stop work");
            }
        }
    }

    @NonNull
    public String getLoplatId() {
        return loplatId;
    }

    @NonNull
    public String getLoplatPw() {
        return loplatPw;
    }

    @NonNull
    public String getHostUrl() {
        return hostUrl;
    }
}

File: app/src/main/java/com/jnjcomu/edison/storage/Configuration.kt
package com.jnjcomu.edison.storage

import android.content.Context
import com.jnjcomu.edison.encrypt.Encrypter
import org.json.JSONObject

/**
 * Configuration Class
 *
 * 키스토어에 의해 암호화된 키를 해독하는 클레스.
 *
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-13
 */
class Configuration(context: Context) {
    var json: JSONObject? = null
        private set

    var loplatId: String? = null
        private set

    var loplatPw: String? = null
        private set

    var hostUrl: String? = null
        private set

    init {
        val encrypt = Encrypter(context)

        context.assets.open(CONFIGURATION_JSON_NAME).use {
            val size = it.available()
            val buffer = ByteArray(size)

            it.read(buffer)
            val readString = String(buffer)

            // get json object
            json = JSONObject(readString)

            // get decrypted key
            val loplatJson = json!!.getJSONObject("loplat")

            val encryptHostUrl = json!!.getString("host_url")
            val encryptLoplatId = loplatJson.getString("id")
            val encryptLoplatPw = loplatJson.getString("pw")

            hostUrl = encrypt.decrypt(encryptHostUrl)
            loplatId = encrypt.decrypt(encryptLoplatId)
            loplatPw = encrypt.decrypt(encryptLoplatPw)
        }
    }

    companion object {
        private val TAG = "Configuration"
        private val ENCODING = "UTF-8"
        private val CONFIGURATION_JSON_NAME = "Edison.json"
    }
}